.github/workflows/check_cirrus_cron.yml

```diff
@@ -0,0 +1,76 @@
+    # Debug: Allow triggering job manually in github-actions WebUI
+    workflow_dispatch: {}
+
+env:
+    # Debug-mode can reveal secrets, only enable by a secret value.
+    # Ref: https://help.github.com/en/actions/configuring-and-managing-workflows/managing-a-workflow-run#enabling-step-debug-logging
+    ACTIONS_STEP_DEBUG: '${{ secrets.ACTIONS_STEP_DEBUG }}'
+    # File with CSV listing of zero or more e-mail addresses for delivery
+    # of daily failure notice e-mails.
+    FAILMAILCSV: './contrib/cirrus/cron_fail_mail.csv'
+
+jobs:
+    cron_failures:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+              with:
+                  ref: master
+                  persist-credentials: false
+
+            - name: Get failed cron names and Build IDs
+              id: cron
+              run: './.github/actions/${{ github.workflow }}/${{ github.job }}.sh'
+
+            - if: steps.cron.outputs.failures > 0
+              shell: bash
+              # Must be inline, since context expressions are used.
+              # Ref: https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions
+              run: |
+                set -eo pipefail
+                (
+                echo "This is your friendly neighborhood ${{ github.action }} github-action on ${{ github.repository }}."
+                echo "I have detected that one or more Cirrus-CI cron-triggered jobs have failed recently:"
+                echo ""yy
+                ) > ./artifacts/email_body.txt
+
+                while read -r NAME BID; do
+                    echo "Cron build '$NAME' Failed: https://cirrus-ci.com/build/$BID"
+                done < ./artifacts/name_id.csv >> ./artifacts/email_body.txt
+                # Make space for sendgrid.com to append it's footer.
+                echo "" >> ./artifacts/email_body.txt
+
+            - if: steps.cron.outputs.failures > 0
+              id: mailto
+              run: printf "::set-output name=csv::%s\n" $(cat "$FAILMAILCSV")
+
+            - if: steps.mailto.outputs.csv != ''
+              name: Send failure notification e-mail
+              # Ref: https://github.com/dawidd6/action-send-mail
+              uses: dawidd6/action-send-mail@v2.2.2
+              with:
+                server_address: ${{secrets.ACTION_MAIL_SERVER}}
+                server_port: 465
+                username: ${{secrets.ACTION_MAIL_USERNAME}}
+                password: ${{secrets.ACTION_MAIL_PASSWORD}}
+                subject: Cirrus-CI cron build failures on ${{github.repository}}
+                to: ${{steps.mailto.outputs.csv}}
+                from: ${{secrets.ACTION_MAIL_SENDER}}
+                body: file://./artifacts/email_body.txt
+
+            - if: always()
+              uses: actions/upload-artifact@v2
+              with:
+                  name: ${{ github.job }}_artifacts
+                  path: artifacts/*
```

Comment:
I assume this is the reason for naming the workdir `artifacts`. The use of a hardcoded path instead of a tmpdir makes me uncomfortable because of concurrency worries (multiple cron jobs stall for a few days, then suddenly all run at once). Do we have a guarantee that these jobs will run on a pristine dedicated-per-job environment? If not, is there a way to call `mktemp` within this yaml and pass that value to the script?

Reply:
Yes, potentially overlapping jobs will never re-use the working directory.  For the artifacts action, the path MUST reside under the workdir, so it can't grab anything under `/tmp/`.

Even if the tempdir is created within the workdir, passing it's name from the shell-script to the `path` value would require jumping through hoops, e.g.:

```bash
    ARTIFACTS=$(mktemp -p "$GITHUB_WORKSPACE" artifacts_XXXXX)
    printf "::set-output name=artifacts::%s\n" "$ARTIFACTS"
    ...
```

then

```yaml
            - if: always()
              uses: actions/upload-artifact@v2
              with:
                  name: ${{ github.job }}_artifacts
                  path: ${{ steps.cron.outputs.artifacts }}/*
```

I can do that if you think it worthwhile...it just seems over-complicated to me.

Reply:
Understood, thanks. Let's leave it as-is then.